resources/user.go

```diff
@@ -0,0 +1,242 @@
+			MetaParams: DefaultMetaParams, // force a default
+		},
+	}
+}
+
+// Validate if the params passed in are valid data.
+func (obj *UserRes) Validate() error {
+	if obj.State != "exists" && obj.State != "absent" {
+		return fmt.Errorf("State must be 'exists' or 'absent'")
+	}
+	return obj.BaseRes.Validate()
+}
+
+// Init initializes the resource.
+func (obj *UserRes) Init() error {
+	return obj.BaseRes.Init() // call base init, b/c we're overriding
+}
+
+// Watch is the primary listener for this resource and it outputs events.
+func (obj *UserRes) Watch() error {
+	var err error
+	obj.recWatcher, err = recwatch.NewRecWatcher("/etc/passwd", false)
+	if err != nil {
+		return err
+	}
+	defer obj.recWatcher.Close()
+
+	// notify engine that we're running
+	if err := obj.Running(); err != nil {
+		return err // bubble up a NACK...
+	}
+
+	var send = false // send event?
+	var exit *error
+
+	for {
+		if obj.debug {
+			log.Printf("Watching: /etc/passwd") // attempting to watch...
+		}
+
+		select {
+		case event, ok := <-obj.recWatcher.Events():
+			if !ok { // channel shutdown
+				return nil
+			}
+			if err := event.Error; err != nil {
+				return errwrap.Wrapf(err, "Unknown %s watcher error", obj)
+			}
+			if obj.debug { // don't access event.Body if event.Error isn't nil
+				log.Printf("%s: Event(%s): %v", obj, event.Body.Name, event.Body.Op)
+			}
+			send = true
+			obj.StateOK(false) // dirty
+
+		case event := <-obj.Events():
+			if exit, send = obj.ReadEvent(event); exit != nil {
+				return *exit // exit
+			}
+			//obj.StateOK(false) // dirty // these events don't invalidate state
+		}
+
+		// do all our event sending all together to avoid duplicate msgs
+		if send {
+			send = false
+			obj.Event()
+		}
+	}
+}
+
+// CheckApply method for User resource. Does nothing, returns happy!
+func (obj *UserRes) CheckApply(apply bool) (checkOK bool, err error) {
+	log.Printf("%s: CheckApply(%t)", obj, apply)
+
+	var exists = true
+
+	usr, err := user.Lookup(obj.GetName())
+
+	log.Printf("%+v", usr)
+
+	if err != nil && err == err.(user.UnknownUserError) {
+		log.Printf("the user: %s does not exist", obj.GetName())
+		exists = false
+	}
+
+	if obj.State == "exists" && exists &&
+		(obj.Gid == "" || obj.Gid == usr.Gid) &&
+		(obj.HomeDir == "" || obj.HomeDir == usr.HomeDir) &&
+		(obj.UID == "" || obj.UID == usr.Uid) {
+		return true, nil
+	}
+
+	if obj.State == "absent" && !exists {
+		return true, nil
+	}
+
+	if !apply {
+		return false, nil
+	}
+
+	var cmd string
+	var args []string
+	if obj.State == "exists" {
+		if !exists {
+			cmd = "useradd"
+		}
+		if exists {
+			cmd = "usermod"
+		}
+		if obj.UID != "" {
+			args = append(args, "-u", obj.UID)
+		}
+		if obj.Gid != "" {
+			args = append(args, "-g", obj.Gid)
+		}
+		if obj.HomeDir != "" {
+			args = append(args, "-d", obj.HomeDir)
+		}
+
+	}
+
+	if obj.State == "absent" {
+		cmd = "userdel"
+	}
+
+	args = append(args, obj.GetName())
+
+	usercmd := exec.Command(cmd, args...)
```

Comment:
You'll need to add these: https://github.com/purpleidea/mgmt/blob/master/resources/exec.go#L206 IIRC